### PR TITLE
#554 - features can use disabled kbs

### PR DESCRIPTION
**Pull request by MdeBoer95**
- add warning label in conecept feature editor
- display warning in concept feature settings

### CODE REVIEW

**Comment by MdeBoer95** [comment]
@reckart How can I add a warning label for the fact linking outside of the ``QualifierFeatureEditor``,``SubjectObjectFeatureEditor``, etc.? 
So that there is only one label at the bottom and not one for each editor.
Is there a Webanno class or something?

**Comment by reckart** [comment]
@MdeBoer95 If I understand the question correctly, I guess you'd have to add the warning label inside the `<div wicket:id="content">` in the HTML files of the two editors.

**Comment by MdeBoer95** [comment]
No, I mean I want to add the label to the surrounding Panel.

**Comment by reckart** [comment]
You could add the marker in the title section of the panel:

```
<div class="panel-heading">
      <h3 class="panel-title">
        <span wicket:id="feature"></span> **HERE**
      </h3>
    </div>
```

... but it would need to be a very small marker. If I saw it right, you currently add a rather longish message explaining the problem. I would suggest instead to use a small warning icon (e.g. https://fontawesome.com/v3.2.1/icon/warning-sign) and put a tooltip on it with the explanatory text.

**Comment by MdeBoer95** [comment]
> I would suggest instead to use a small warning icon (e.g. https://fontawesome.com/v3.2.1/icon/warning-sign) and put a tooltip on it with the explanatory text.

makes sense

> You could add the marker in the title section of the panel:

But wouldn't I need to add the Label/Icon for every Editor in that case?
If I just use a small Icon it might be okay. I just wanted to avoid displaying a long message multiple times.

**Comment by reckart** [comment]
> But wouldn't I need to add the Label/Icon for every Editor in that case?
If I just use a small Icon that might be okay. I just wanted to avoid displaying a long message multiple times.

Right - every editor to which the problem applies would exhibit a warning. Mind that an annotation could have multiple editors, some to which the problem applies and some to which it does not apply.

**Comment by reckart** [comment]
The issue number referenced in the commits has a typo - it should be #554, not #544

File: inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ConceptFeatureEditor.java [hunk 1]
```diff
@@ -165,6 +175,25 @@ private JCas getEditorCas(AnnotationActionHandler aHandler) throws IOException {
         return handles;
     }
 
+    private boolean featureUsesDisabledKB(ConceptFeatureTraits aTraits)
+    {
+        Optional<KnowledgeBase> kb = null;
```
Comment (Rentier) [inline inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/ConceptFeatureEditor.java]: Please do not assign `null` to an `Optional`, use `Optional.empty()`

**Comment by naveen2507** [comment]
Jenkins, can you test this please?

**Comment by naveen2507** [comment]
@MdeBoer95 I dont get any warning message, while I look over the annotations from the KB which has been disabled. Although debugging shows that it goes through  `add(createDisabledKbWarningLabel());` in  `ConceptFeatureEditor.java`

**Comment by naveen2507** [comment]
Works.

**Pull request merged**